File: neerc.ifmo.ru.train.2014.10.28/F.java
import java.util.*;
import java.io.*;
 
public class F {
 
    long sum(long[] p, int l, int r) {
        if (r >= p.length) {
            r = p.length - 1;
        }
 
        long ans = p[r];
        if (l > 0) {
            ans -= p[l - 1];
        }
        return ans;
 
    }
 
    public void solve() {
        int n = in.nextInt(), k = in.nextInt();
 
        long[] p = new long[n];
        p[0] = in.nextInt();
        for (int i = 1; i < n; i++) {
            p[i] = p[i - 1] + in.nextInt();
        }
 
        if (k * 2 >= n) {
            out.println(p[n - 1]);
            return;
        }
 
        long[] m = new long[n];
 
        long q = 0;
 
        for (int i = n - 1; i >= 0; i--) {
            q = Math.max(q, sum(p, i, i + k - 1));
            m[i] = q;
        }
 
        long ans = 0;
 
        for (int i = 0; i < n; i++) {
            long cur = sum(p, i, i + k - 1);
            if (i + k < n) {
                cur += m[i + k];
            }
            ans = Math.max(ans, cur);
        }
 
        out.println(ans);
    }
 
    FastScanner in;
    PrintWriter out;
 
    public void run() {
        in = new FastScanner();
        out = new PrintWriter(System.out);
 
        solve();
 
        out.close();
    }
 
    class FastScanner {
        BufferedReader br;
        StringTokenizer st;
 
        FastScanner() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }
 
        String nextLine() {
            try {
                return br.readLine();
            } catch (IOException e) {
                return null;
            }
        }
 
        String next() {
            while (st == null || !st.hasMoreTokens()) {
                st = new StringTokenizer(nextLine());
            }
            return st.nextToken();
        }
 
        int nextInt() {
            return Integer.parseInt(next());
        }
    }
 
    public static void main(String[] arg) {
        new F().run();
    }
}

File: CodeForce/1857/F.py
import math

def solve(b, c):
    d = b * b - 4 * c
    if d < 0:
        return None
    s = int(math.sqrt(d))
    while (s * s < d): s += 1
    while (s * s > d): s -= 1
    if s * s != d:
        return None
    q = -b + s
    w = -b - s
    if q % 2 != 0:
        return None
    if w % 2 != 0:
        return None
    return [-q // 2, -w // 2]


def run():
    n = int(input())
    a = list(map(int, input().split()))
    D = {}
    for x in a:
        D[x] = D.get(x, 0) + 1

    ans = []
    q = int(input())
    for Q in range(q):
        b, c = list(map(int, input().split()))
        cur = solve(b, c)
        if cur is None:
            ans.append(0)
        else:
            x, y = cur
            w = D.get(x, 0)
            e = D.get(y, 0)
            if x == y:
                ans.append(w * (w - 1) // 2)
            else:
                ans.append(w * e)
    print(*ans)

t = int(input())
for i in range(t):
    run()
